### PR TITLE
[DependencyInjection] make the service container builder register its own self referencing definition

### PR DESCRIPTION
**Pull request by hhamon**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ~
| License       | MIT
| Doc PR        | ~

### CODE REVIEW

**Comment by hhamon** [comment]
This PR fixes a bug in Symfony 3.3 that prevents autowiring compiler passes to autowire the service container in a service. The bug mostly occurs when you use `setter` methods autowiring when defining regular Symfony controller classes as services.

```yaml
services:

    AppBundle\Controller\:
        resource: '../../src/AppBundle/Controller'
        autowire: ['get*', 'set*']
```

If the controller class still extends the base Symfony controller class (or implements the `ContainerAwareInterface`), the service container will be injected through the `setContainer` autowired method.

**Comment by hhamon** [comment]
Looks like fabbot.io violations is a false positive due to a fixtures file,

**Comment by hhamon** [comment]
@nicolas-grekas can you review please.

**Review by nicolas-grekas — APPROVED** [review]
:+1:

**Review by xabbuh — APPROVED** [review]
:+1:

**Comment by nicolas-grekas** [comment]
Just wondering: should we also add these three aliases to the defaults also?
```xml
        <service id="Psr\Container\ContainerInterface" alias="service_container" public="false" />
        <service id="Symfony\Component\DependencyInjection\ContainerInterface" alias="service_container" public="false" />
        <service id="Symfony\Component\DependencyInjection\Container" alias="service_container" public="false" />
```

**Comment by hhamon** [comment]
What do you mean?

**Comment by nicolas-grekas** [comment]
I mean add three cal behind the new one, to `setAlias(ContainerInterface::class/etc, new Alias('servicecontainer', false)`

**Comment by hhamon** [comment]
I'm on it.

**Comment by hhamon** [comment]
@nicolas-grekas better?

**Comment by nicolas-grekas** [comment]
yes! you can also clean `web.xml` in FrameworkBundle and that will be good to me.

**Comment by hhamon** [comment]
Shall we use autowire in `web.xml`? I can see only one reference to the service container that is injected into the `ControllerResolver` object.

**Comment by hhamon** [comment]
I think you're talking about the `services.xml` file in FrameworkBundle to cleanup.

**Comment by hhamon** [comment]
@nicolas-grekas done ;)

**Comment by nicolas-grekas** [comment]
Missing test case fix:
https://travis-ci.org/symfony/symfony/jobs/203205123#L3297

**Comment by hhamon** [comment]
Review from Symfony Core Team member is welcome.

**Comment by xabbuh** [comment]
:+1: for the added aliases too

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services10.php [hunk 1]
```diff
@@ -30,6 +30,11 @@ public function __construct()
         $this->parameters = $this->getDefaultParameters();
 
         $this->services = array();
+        $this->normalizedIds = array(
+            'psr\\container\\containerinterface' => 'Psr\\Container\\ContainerInterface',
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services10.php]: @nicolas-grekas should we really dump normalized ids for services which got removed ?

Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services10.php]: dunno :) do we really care? if someone wants to try fixing that, please do :) should be done while dumping - ie in PhpDumper I'd say.

**Comment by fabpot** [comment]
Thank you @hhamon.

**Pull request merged**